### PR TITLE
HF: Move block.proof.challenge to Consensus::Params::signblockScript

### PR DESCRIPTION
**Pull request by jtimon**
Like #114 but rebased to 0.14.

Dependencies:

- [x] HF: Genesis block includes fedpegscript #187
- [x] Remove CChainParams::bitcoinID #188

### CODE REVIEW

**Comment by jtimon** [comment]
Rebased after #187 was merged for clarity.

**Comment by jtimon** [comment]
Needed rebase.

File: src/chainparams.cpp [hunk 1]
```diff
@@ -52,16 +52,17 @@ static CBlock CreateGenesisBlock(const Consensus::Params& params, const std::str
     txNew.vin.resize(1);
     txNew.vout.resize(rewardShards);
     // Any consensus-related values that are command-line set can be added here for anti-footgun
-    txNew.vin[0].scriptSig = CScript(CommitToArguments(params, networkID, scriptChallenge));
+    txNew.vin[0].scriptSig = CScript(CommitToArguments(params, networkID));
+    txNew.vout.resize(rewardShards);
```
Comment (jtimon) [inline src/chainparams.cpp]: Duplicated line

**Comment by jtimon** [comment]
Fixed the auto-nit: https://travis-ci.org/ElementsProject/elements/jobs/268547573 was failing already. Yay travis!

**Comment by jtimon** [comment]
Needed rebase

**Comment by jonasnick** [comment]
concept ACK

**Comment by instagibbs** [comment]
concept ACK as well. I think it might be a good time to work on merging this.

**Comment by jtimon** [comment]
Some tests are failing. I will look it up (but if anyone else helps fixing them, that would be great).

**Comment by jtimon** [comment]
Rebased

**Comment by jtimon** [comment]
Rebased

**Comment by instagibbs** [comment]
bunch of tests failing, haven't looked into it

**Comment by jtimon** [comment]
it seems sync_all is timing out for those tests, looks like a bug.

**Comment by jtimon** [comment]
Closed by accident, reopened as https://github.com/ElementsProject/elements/pull/313

**Pull request closed without merging**